### PR TITLE
[Docs Site] Change all internal links to relative

### PR DESCRIPTION
**Pull request by KianNH**
Changes absolute internal links to be relative - mainly because they're annoying when on preview links (or forks) as they take you away and onto the main site but partly because I think the style guide wants them to be relative anyways.

Please spot-check for any malformed links. This commit doesn't include other links, like sub notes, just ones that are part of a Markdown link.

### CODE REVIEW

**Comment by KianNH** [comment]
This branch was built at https://kian-relative-links.cloudflare-docs-github.pages.dev and can be used for testing links - I've tested some API links to ensure they work the same as the production site.

**Comment by KianNH** [comment]
I now understand why the `/api/` links wasn't relative - because the link checker can't validate them in CI.

**Comment by kodster28** [comment]
Let me take a bit of time later today and push up some logic for this.

Already doing it with some of our links in the `/support/` tile, so it shouldn't be that hard to do the same for anything starting with `/api/`.

**Comment by kodster28** [comment]
@KianNH, if you merge in production, this issue should be fixed.

File: content/d1/learning/backups.md [hunk 1]
```diff
@@ -69,7 +69,7 @@ $ wrangler d1 backup download example-db 123a81a2-ab91-4c2e-8ebc-64d69633faf1
 🌀 Done!
 ```
 
-The database backup will be download to the current working directory in native SQLite3 format. To import a local database, read [the documentation on importing data](https://developers.cloudflare.com//learning/importing-data/) to D1.
+The database backup will be download to the current working directory in native SQLite3 format. To import a local database, read [the documentation on importing data](/learning/importing-data/) to D1.
```
Comment (KianNH) [inline content/d1/learning/backups.md]:
```suggestion
The database backup will be download to the current working directory in native SQLite3 format. To import a local database, read [the documentation on importing data](/d1/learning/importing-data/) to D1.
```

Not sure how this link wasn't picked up by the crawler originally 🤔

File: content/queues/learning/batching-retries.md [hunk 2]
```diff
@@ -84,7 +84,7 @@ When a message is failed to be delivered, the default behaviour is to retry deli
 
 {{<Aside type="note">}}
 
-Each retry counts as an additional read operation per [Queues pricing](https://developers.cloudflare.com/queues/pricing).
+Each retry counts as an additional read operation per [Queues pricing](/queues/pricing).
```
Comment (KianNH) [inline content/queues/learning/batching-retries.md]:
```suggestion
Each retry counts as an additional read operation per [Queues pricing](/queues/platform/pricing/).
```

File: content/support/Troubleshooting/Cloudflare Errors/Troubleshooting Cloudflare 1XXX errors.md [hunk 3]
```diff
@@ -63,15 +63,15 @@ ___
 -   An external domain that is not on using Cloudflare has a CNAME record to a domain active on Cloudflare
 -   The target of the DNS CNAME record does not resolve.
 -   A CNAME record in your Cloudflare DNS app requires resolution via a DNS provider that is currently offline.
--   [Always Online](https://developers.cloudflare.com/cache/about/always-online) is enabled for a [Custom Hostname (SSL for SaaS](https://developers.cloudflare.com/ssl/ssl-for-saas)) domain.
+-   [Always Online](/cache/about/always-online) is enabled for a [Custom Hostname (SSL for SaaS](/ssl/ssl-for-saas)) domain.
```
Comment (KianNH) [inline content/support/Troubleshooting/Cloudflare Errors/Troubleshooting Cloudflare 1XXX errors.md]:
```suggestion
-   [Always Online](/cache/about/always-online) is enabled for a [Custom Hostname (Cloudflare for SaaS](/cloudflare-for-platforms/cloudflare-for-saas/)) domain.
```

File: content/support/Troubleshooting/Cloudflare Errors/Troubleshooting Cloudflare 1XXX errors.md [hunk 4]
```diff
@@ -63,15 +63,15 @@ ___
...
 
 ### Resolution
 
 A non-Cloudflare domain cannot CNAME to a Cloudflare domain unless the non-Cloudflare domain is added to a Cloudflare account.
 
-Attempting to directly access DNS records used for [Cloudflare CNAME setups](https://developers.cloudflare.com/dns/zone-setups/partial-setup) also causes error 1001 (For example: _www.example.com.cdn.cloudflare.net_).
+Attempting to directly access DNS records used for [Cloudflare CNAME setups](/dns/zone-setups/partial-setup) also causes error 1001 (For example: _www.example.com.cdn.cloudflare.net_).
 
-Disable [Always Online](https://developers.cloudflare.com/cache/how-to/enable-always-online) if using [Custom Hostnames (SSL for SaaS)](https://developers.cloudflare.com/ssl/ssl-for-saas).
+Disable [Always Online](/cache/how-to/enable-always-online) if using [Custom Hostnames (SSL for SaaS)](/ssl/ssl-for-saas).
```
Comment (KianNH) [inline content/support/Troubleshooting/Cloudflare Errors/Troubleshooting Cloudflare 1XXX errors.md]:
```suggestion
Disable [Always Online](/cache/how-to/enable-always-online) if using [Custom Hostname (Cloudflare for SaaS](/cloudflare-for-platforms/cloudflare-for-saas/)) domain.
```

File: content/support/Troubleshooting/Cloudflare Errors/Troubleshooting Cloudflare 1XXX errors.md [hunk 5]
```diff
@@ -155,7 +155,7 @@ If you are not the website owner, provide the website owner with a screenshot of
 If you are the website owner:
 
 1.  Retrieve a screenshot of the 1005 error from your customer
-2.  Search the [**Security Events log**](https://developers.cloudflare.com/waf/analytics) (available at **Security** \> **Events**) for the **RayID**, or client IP Address from the visitor’s 1005 error message.
+2.  Search the [**Security Events log**](/waf/analytics) (available at **Security** \> **Events**) for the **RayID**, or client IP Address from the visitor’s 1005 error message.
```
Comment (KianNH) [inline content/support/Troubleshooting/Cloudflare Errors/Troubleshooting Cloudflare 1XXX errors.md]:
```suggestion
2.  Search the [**Security Events log**](/waf/security-events/) (available at **Security** \> **Events**) for the **RayID**, or client IP Address from the visitor’s 1005 error message.
```

File: content/support/Troubleshooting/Cloudflare Errors/Troubleshooting Cloudflare 1XXX errors.md [hunk 6]
```diff
@@ -275,14 +275,14 @@ By default, Cloudflare prohibits a DNS _CNAME record_ between domains in diffe
 {{<Aside type="warning">}}
 [Cloudflare Apps](https://www.cloudflare.com/apps/) are not currently
 supported by [SSL for
```
Comment (KianNH) [inline content/support/Troubleshooting/Cloudflare Errors/Troubleshooting Cloudflare 1XXX errors.md]:
```suggestion
supported by [Cloudflare for
```

File: content/support/Troubleshooting/Cloudflare Errors/Troubleshooting Cloudflare 1XXX errors.md [hunk 7]
```diff
@@ -275,14 +275,14 @@ By default, Cloudflare prohibits a DNS _CNAME record_ between domains in diffe
...
-SaaS](https://developers.cloudflare.com/ssl/ssl-for-saas/), therefore
+SaaS](/ssl/ssl-for-saas/), therefore
```
Comment (KianNH) [inline content/support/Troubleshooting/Cloudflare Errors/Troubleshooting Cloudflare 1XXX errors.md]:
```suggestion
SaaS](/cloudflare-for-platforms/cloudflare-for-saas/), therefore
```

File: content/support/Troubleshooting/Cloudflare Errors/Troubleshooting Cloudflare 1XXX errors.md [hunk 8]
```diff
@@ -275,14 +275,14 @@ By default, Cloudflare prohibits a DNS _CNAME record_ between domains in diffe
...
 any app using a domain configured on our SaaS solution may produce 1014
 errors.
 {{</Aside>}}
 
 ### Resolution
 
-To allow CNAME record resolution to a domain in a different Cloudflare account, the domain owner of the CNAME target must use [Cloudflare for SaaS](https://www.cloudflare.com/saas/); more specially our [SSL for SaaS](https://developers.cloudflare.com/ssl/ssl-for-saas/) solution.
+To allow CNAME record resolution to a domain in a different Cloudflare account, the domain owner of the CNAME target must use [Cloudflare for SaaS](https://www.cloudflare.com/saas/); more specially our [SSL for SaaS](/ssl/ssl-for-saas/) solution.
```
Comment (KianNH) [inline content/support/Troubleshooting/Cloudflare Errors/Troubleshooting Cloudflare 1XXX errors.md]:
```suggestion
To allow CNAME record resolution to a domain in a different Cloudflare account, the domain owner of the CNAME target must use [Cloudflare for SaaS](/cloudflare-for-platforms/cloudflare-for-saas/).
```

File: content/support/Troubleshooting/Cloudflare Errors/Troubleshooting Cloudflare 1XXX errors.md [hunk 9]
```diff
@@ -326,8 +326,8 @@ Common causes for Error 1016 are:
 
 -   A missing DNS _A record_ that mentions origin IP address.
 -   A _CNAME record_ in the Cloudflare DNS points to an unresolvable external domain.
--   The origin host names (CNAMEs) in your Cloudflare [Load Balancer](https://developers.cloudflare.com/load-balancing/) default, region, and fallback pools are unresolvable. Use a fallback pool configured with an origin IP as a backup in case all other pools are unavailable.
--   When creating a Spectrum app with a CNAME origin, you need first to create a CNAME on the Cloudflare DNS side that points to the origin. Please see [Spectrum CNAME origins](https://developers.cloudflare.com/spectrum/how-to/cname-origins) for more details
+-   The origin host names (CNAMEs) in your Cloudflare [Load Balancer](/load-balancing/) default, region, and fallback pools are unresolvable. Use a fallback pool configured with an origin IP as a backup in case all other pools are unavailable.
+-   When creating a Spectrum app with a CNAME origin, you need first to create a CNAME on the Cloudflare DNS side that points to the origin. Please see [Spectrum CNAME origins](/spectrum/how-to/cname-origins) for more details
```
Comment (KianNH) [inline content/support/Troubleshooting/Cloudflare Errors/Troubleshooting Cloudflare 1XXX errors.md]:
```suggestion
-   When creating a Spectrum app with a CNAME origin, you need first to create a CNAME on the Cloudflare DNS side that points to the origin. Please see [Spectrum CNAME origins](/spectrum/get-started/#create-a-spectrum-application-using-a-cname-record) for more details
```

File: content/support/Troubleshooting/Cloudflare Errors/Troubleshooting Cloudflare 1XXX errors.md [hunk 10]
```diff
@@ -384,7 +384,7 @@ If you are not the website owner, provide the website owner with a screenshot of
 If you are the website owner:
 
 1.  Retrieve a screenshot of the 1020 error from your customer
-2.  Search the [**Security Events log**](https://developers.cloudflare.com/waf/analytics) (available at **Security** \> **Events**) for the **RayID** or client IP Address from the visitor’s 1020 error message.
+2.  Search the [**Security Events log**](/waf/analytics) (available at **Security** \> **Events**) for the **RayID** or client IP Address from the visitor’s 1020 error message.
```
Comment (KianNH) [inline content/support/Troubleshooting/Cloudflare Errors/Troubleshooting Cloudflare 1XXX errors.md]:
```suggestion
2.  Search the [**Security Events log**](/waf/security-events/) (available at **Security** \> **Events**) for the **RayID** or client IP Address from the visitor’s 1020 error message.
```

File: content/support/Troubleshooting/Cloudflare Errors/Troubleshooting Cloudflare 1XXX errors.md [hunk 11]
```diff
@@ -511,7 +511,7 @@ You are trying to modify an HTTP header that HTTP Request Header Modification Ru
 
 ### Resolution
 
-Make sure you are not trying to modify one of the [reserved HTTP request headers](https://developers.cloudflare.com/rules/transform#http-request-header-modification-rules).
+Make sure you are not trying to modify one of the [reserved HTTP request headers](/rules/transform#http-request-header-modification-rules).
```
Comment (KianNH) [inline content/support/Troubleshooting/Cloudflare Errors/Troubleshooting Cloudflare 1XXX errors.md]:
```suggestion
Make sure you are not trying to modify one of the [reserved HTTP request headers](/rules/transform/request-header-modification/#important-remarks).
```

File: content/support/Troubleshooting/Cloudflare Errors/Troubleshooting Cloudflare 1XXX errors.md [hunk 12]
```diff
@@ -524,7 +524,7 @@ The added/modified header value is too long or it contains characters that are n
 ### Resolution
 
 -   Use a shorter value or expression to define the header value.
--   Remove the characters that are not allowed. See [Format of HTTP request header names and values](https://developers.cloudflare.com/rules/transform/create-header-modification-rule#format-of-http-request-header-names-and-values) in Developer Docs for more information on the allowed characters.
+-   Remove the characters that are not allowed. See [Format of HTTP request header names and values](/rules/transform/create-header-modification-rule#format-of-http-request-header-names-and-values) in Developer Docs for more information on the allowed characters.
```
Comment (KianNH) [inline content/support/Troubleshooting/Cloudflare Errors/Troubleshooting Cloudflare 1XXX errors.md]:
```suggestion
-   Remove the characters that are not allowed. See [Format of HTTP request header names and values](/rules/transform/request-header-modification/reference/header-format/) in Developer Docs for more information on the allowed characters.
```

**Comment by kodster28** [comment]
We'll spot-check review this tomorrow, btw.

**Comment by KianNH** [comment]
Sounds good to me - there's a few ones that I'm a little confused about (i.e D1 & Queues) where the links have always been invalid, but not picked up.

Does it only validate relative links & assumes absolute links are external?

I cleaned up a few links in the support category too - mostly the troubleshooting 1xxx page.

**Review by kodster28 — APPROVED** [review]

**Comment by kodster28** [comment]
Haha, and by tomorrow, I meant during this lunch break.

Yeah, our link validation doesn't evaluate absolute URLs (we have an SEO crawler tool that does that), so it only "realized" they were broken once they became relative.

It also ignores redirects for easier maintenance down the road.

In a future PR, I'll also re-add the logic to add warnings about full developers.cloudflare.com URLs to the Compiles step.

**Pull request merged**